### PR TITLE
Removes inline styling

### PR DESCRIPTION
**Pull request by idealadarsh**
Eliminates inline styling from index.html.in
Adds HTML Lint rule "inline-style-disabled": true
Moves all inline styling code to assets/css/style.css

Related to Issue https://github.com/goavki/apertium-html-tools/issues/114

Done according to: https://codein.withgoogle.com/dashboard/task-instances/5437388455149568/

### CODE REVIEW

File: assets/css/style.css [hunk 1]
```diff
@@ -396,3 +396,108 @@ html[dir='rtl'] .pull-right:not(#rtlChecker) { /* sass-lint:disable-line class-n
...
+.posRelative {
+    position: relative;
+}
+#apertiumLogo {
+    margin-bottom: -.25em;
+}
+#localeSelectDiv {
+    width: 35%;
+}
+#localeGlobeIcon{
```
Comment (sushain97) [inline assets/css/style.css]: There should be spacing between the selector an initial brace.

File: assets/css/style.css [hunk 2]
```diff
@@ -396,3 +396,108 @@ html[dir='rtl'] .pull-right:not(#rtlChecker) { /* sass-lint:disable-line class-n
...
+    padding: 5px 5px 0 0;
+}
+#srcLangSelect, #dstLangSelect {
```
Comment (sushain97) [inline assets/css/style.css]: There should be newlines between each rule.

File: assets/css/style.css [hunk 3]
```diff
@@ -396,3 +396,108 @@ html[dir='rtl'] .pull-right:not(#rtlChecker) { /* sass-lint:disable-line class-n
 html[dir='rtl'] .pull-left:not(#rtlChecker) { /* sass-lint:disable-line class-name-format */
     float: right !important; /* sass-lint:disable-line no-important */
 }
+.posRelative {
```
Comment (sushain97) [inline assets/css/style.css]: These rules should be organized into the correct sections in this stylesheet as per the comments, e.g. none of this belongs under the `/* RTL Stuff */` section.

File: assets/css/style.css [hunk 4]
```diff
@@ -396,3 +396,108 @@ html[dir='rtl'] .pull-right:not(#rtlChecker) { /* sass-lint:disable-line class-n
...
+    cursor: pointer;
+}
+#GSocImg{
+    height: 2.5em;
+}
+#GCIImg{
+    height: 2.5em;
+    padding-left: .5em;
+}
+.coltype1{
```
Comment (sushain97) [inline assets/css/style.css]: What is `.coltype1`? I suggest using more descriptive names for these.

File: assets/css/style.css [hunk 5]
```diff
@@ -396,3 +396,108 @@ html[dir='rtl'] .pull-right:not(#rtlChecker) { /* sass-lint:disable-line class-n
...
+#localeGlobeIconXs{
+    padding: 5px 0 0 5px;
+}
+.inlinedisplay{
+    display: inline-block;
+}
+#helpMessage{
+    margin-top: 10px;
+}
+#Maintainer{
```
Comment (sushain97) [inline assets/css/style.css]: These should be consistent in case, i.e. camelCase.

File: assets/css/style.css [hunk 6]
```diff
@@ -396,3 +396,108 @@ html[dir='rtl'] .pull-right:not(#rtlChecker) { /* sass-lint:disable-line class-n
...
+    display: none;
+    z-index: 1000;
+}
+#modalWell{
+    left: 35%;
+    position: absolute;
+    text-align: center;
+    top: 35%;
+}
+.bold{
```
Comment (sushain97) [inline assets/css/style.css]: This doesn't really warrant its own class.

File: assets/css/style.css [hunk 7]
```diff
@@ -396,3 +396,108 @@ html[dir='rtl'] .pull-right:not(#rtlChecker) { /* sass-lint:disable-line class-n
...
+#localeSelectDiv {
+    width: 35%;
+}
+#localeGlobeIcon{
+    padding: 5px 5px 0 0;
+}
+#srcLangSelect, #dstLangSelect {
+    display: inline-block;
+}
+#cancelTranslateWebpage{
```
Comment (sushain97) [inline assets/css/style.css]: Why not add `.nodisplay` instead?

File: assets/css/style.css [hunk 8]
```diff
@@ -396,3 +396,108 @@ html[dir='rtl'] .pull-right:not(#rtlChecker) { /* sass-lint:disable-line class-n
...
+#fileUploadProgress{
+    width: 0%;
+}
+#morphAnalyzerOutput{
+    border-spacing: 0;
+}
+#localeGlobeIconXs{
+    padding: 5px 0 0 5px;
+}
+.inlinedisplay{
```
Comment (sushain97) [inline assets/css/style.css]: I suggest `.display-inline` instead as per Bootstrap's typical naming convention.

**Comment by sushain97** [comment]
Generally, I suggest taking inspiration for class names from Bootstrap 4's docs. e.g. https://getbootstrap.com/docs/4.0/utilities/text/

**Comment by idealadarsh** [comment]
@sushain97 Ok, I changed some code.
Thanks for the review.

File: assets/css/style.css [hunk 9]
```diff
@@ -103,6 +103,144 @@ html[dir='rtl'] .clearButton {
...
+
+#maintainer {
+    padding-bottom: 2em;
+}
+
+.nodecoration {
+    text-decoration: none;
+}
+
+#BytemarkImg {
```
Comment (shardulc) [inline assets/css/style.css]: camelCase everywhere is preferred, not CapitalCase, even for proper nouns

**Comment by shardulc** [comment]
@idealadarsh @sushain97 Other than my comment above, looks good to me.

File: assets/css/style.css [hunk 10]
```diff
@@ -259,19 +398,17 @@ html[dir='rtl'] #srcLanguages {
...
-   padding-left: 0;
+    padding-left: 0;
 }
 
 @media(min-width: 768px) {
     .swapLangBtn {
         position: absolute;
         top: 0;
     }
-
```
Comment (sushain97) [inline assets/css/style.css]: Why did you delete this standard spacing between rulesets?

File: assets/css/style.css [hunk 11]
```diff
@@ -331,33 +471,38 @@ html[dir='rtl'] #srcLanguages {
     margin-top: 4em;
 }
 
-body > .container {
+body>.container {
```
Comment (sushain97) [inline assets/css/style.css]: Why did you remove this standard binary spacing?

**Comment by idealadarsh** [comment]
@shardulc Changed that

**Comment by shardulc** [comment]
I meant all such instances, not just that one :P

File: assets/css/style.css [hunk 12]
```diff
@@ -196,6 +334,7 @@ html[dir='rtl'] .apertiumLogo {
     }
 }
 
+
```
Comment (sushain97) [inline assets/css/style.css]: I'm not sure that adding all this extra space adds much. Could you explain your reasoning?

File: assets/css/style.css [hunk 13]
```diff
@@ -137,7 +275,7 @@ html[dir='rtl'] .clearButton {
     top: .7em;
 }
 
-.apertiumSubLogo + .apertiumLogo {
+.apertiumSubLogo+.apertiumLogo {
```
Comment (sushain97) [inline assets/css/style.css]: Why did you remove this standard binary spacing?

File: assets/css/style.css [hunk 14]
```diff
@@ -259,19 +398,17 @@ html[dir='rtl'] #srcLanguages {
 }
 
 #srcLangSelectors {
-   padding-left: 0;
+    padding-left: 0;
```
Comment (sushain97) [inline assets/css/style.css]: Good catch.

File: assets/css/style.css [hunk 15]
```diff
@@ -331,33 +471,38 @@ html[dir='rtl'] #srcLanguages {
...
+
 #wrap {
     height: auto !important; /* sass-lint:disable-line no-important */
     margin: 0 auto -60px; /* Negative indent footer by it's height */
     min-height: 99.5%;
 }
 
 #push {
-    height: 60px; /* Set the fixed height of the footer */
+    height: 60px;  /* Set the fixed height of the footer */
```
Comment (sushain97) [inline assets/css/style.css]: ?

File: index.html.in [hunk 16]
```diff
@@ -217,22 +217,22 @@
                                   </div>
                               </div>
 
-                              <div class="modal-backdrop" id="fileDropBackdrop" style="display: none; z-index: 1000;">
-                                <div class="well well-lg" style="text-align: center; position: absolute; top: 35%; left: 35%;">
-                                    <h1 style="font-weight: bold;" data-text="Drop_Document">Drop your document</h1>
+                              <div class="modal-backdrop" id="fileDropBackdrop">
+                                <div class="well well-lg" id="modalWell">
+                                    <h1 id="dropDocHeader" data-text="Drop_Document">Drop your document</h1>
```
Comment (sushain97) [inline index.html.in]: Instead of giving all three elements here unique `id`s, its much more preferable to use `. fileDropBackdrop h1`, `. fileDropBackdrop .well`, etc. This same comment might apply in a couple of other places.

File: index.html.in [hunk 17]
```diff
@@ -217,22 +217,22 @@
...
                                 </div>
-                                <div class="modal-backdrop" id="fileDropMask" style="opacity: 0; z-index: 10000;"></div>
+                                <div class="modal-backdrop" id="fileDropMask"></div>
                               </div>
 
-                              <div id="docTranslation" style="display: none;">
+                              <div id="docTranslation" class="noDisplay">
```
Comment (sushain97) [inline index.html.in]: How do you feel about switching `noDisplay` with `hide`? The latter is much more 'Bootstrapy'.

File: index.html.in [hunk 18]
```diff
@@ -468,13 +468,13 @@
...
                         <div class="row">
-                            <div style="display: inline-block; margin: 0 .5em 0 .5em">
+                            <div class="display-inline halfEmMargin">
                                 <a href="http://creativecommons.org/licenses/by-sa/3.0/">
                                     <img alt="Creative Commons licence" data-src="./img/cc-by-sa-3.0-88x31.png">
                                 </a>
                             </div>
 
-                            <div style="display: inline-block; margin: 0 .5em 0 .5em">
+                            <div class="display-inline halfEmMargin">
```
Comment (sushain97) [inline index.html.in]: Classes shouldn't be made for super specific situations like `. halfEmMargin`, prefer `id`s in this case.

Comment (sushain97) [reply index.html.in]: Decent rule of thumb: if you're not gonna use it again, probably not a good candidate for a class.

**Comment by idealadarsh** [comment]
@shardulc  @sushain97  Changed all of them

File: assets/css/style.css (hunk 14 above)
Comment (shardulc) [reply assets/css/style.css]: The style is aptly 'padding-left'...

File: assets/css/style.css (hunk 13 above)
Comment (idealadarsh) [reply assets/css/style.css]: We should really create a .editorconfig file. I just used VSCode's beautify plugin to do that.

Comment (sushain97) [reply assets/css/style.css]: Feel free to do it and send a PR.

File: index.html.in [hunk 19]
```diff
@@ -559,7 +559,7 @@
 
         <div id="noscript">
             <div class="lead text-warning well well-sm" data-text="Enable_JS_Warning"></div>
-            <noscript><p><img src="@include_piwik_url@/piwik.php?idsite=@include_piwik_siteid@" style="border: 0;" alt=""></p></noscript>
+            <noscript><p><img src="@include_piwik_url@/piwik.php?idsite=@include_piwik_siteid@" class="noborder" alt=""></p></noscript>
```
Comment (sushain97) [inline index.html.in]: https://getbootstrap.com/docs/4.0/utilities/borders/ `border-0` instead

File: index.html.in [hunk 20]
```diff
@@ -547,9 +547,9 @@
                         </p>
                         <p data-text="Contact_Us"></p>
 
-                        <div class="center-block" style="margin-bottom: 1.5em;">
+                        <div class="center-block" id="apertiumImgContainer">
                             <a href="http://www.apertium.org/" target="_blank">
-                                <img alt="Apertium" data-src="./img/Apertium.png" class="center-block" style="height: 5em; max-width: 100%;">
+                                <img alt="Apertium" data-src="./img/Apertium.png" class="center-block" id="apertiumImg">
```
Comment (sushain97) [inline index.html.in]: `#apertiumImgContainer img`

File: index.html.in [hunk 21]
```diff
@@ -411,28 +411,28 @@
...
                                 <div class="center-block">
-                                    <a href="https://developers.google.com/open-source/soc/" target="_blank" style="text-decoration: none">
-                                        <img alt="Google Summer of Code" data-src="https://summerofcode.withgoogle.com/static/img/summer-of-code-logo.svg" style="height: 2.5em;">
+                                    <a href="https://developers.google.com/open-source/soc/" target="_blank" class="nodecoration">
+                                        <img alt="Google Summer of Code" data-src="https://summerofcode.withgoogle.com/static/img/summer-of-code-logo.svg" id="gsocImg">
                                     </a>
-                                    <a href="https://developers.google.com/open-source/gci/" target="_blank" style="text-decoration: none">
-                                        <img alt="Google Code-In" data-src="https://developers.google.com/open-source/gci/images/logo-icon.png" style="height: 2.5em; padding-left: .5em;">
+                                    <a href="https://developers.google.com/open-source/gci/" target="_blank" class="nodecoration">
```
Comment (sushain97) [inline index.html.in]: `no-decoration` in the Utilities section I mentioned below? Sorry, I read this file backwards.

File: index.html.in [hunk 22]
```diff
@@ -217,22 +217,22 @@
...
 
-                              <div class="modal-backdrop" id="fileDropBackdrop" style="display: none; z-index: 1000;">
-                                <div class="well well-lg" style="text-align: center; position: absolute; top: 35%; left: 35%;">
-                                    <h1 style="font-weight: bold;" data-text="Drop_Document">Drop your document</h1>
+                              <div class="modal-backdrop" id="fileDropBackdrop">
+                                <div class="well well-lg">
+                                    <h1 data-text="Drop_Document">Drop your document</h1>
                                 </div>
-                                <div class="modal-backdrop" id="fileDropMask" style="opacity: 0; z-index: 10000;"></div>
+                                <div class="modal-backdrop" id="fileDropMask"></div>
```
Comment (sushain97) [inline index.html.in]: Where is `#fileDropMask` coming from?

File: index.html.in [hunk 23]
```diff
@@ -85,13 +85,13 @@
                       <div>
                           <img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" alt="Apertium Box" class="apertiumBox">
                           <span class="apertiumSubLogo"></span>
-                          <span class="apertiumLogo" style="margin-bottom: -.25em">Apertium</span>
+                          <span class="apertiumLogo" id="apertiumLogo">Apertium</span>
                       </div>
                       <p data-text="tagline" class="tagline"></p>
                   </div>
-                  <div style="width: 35%" class="pull-right hidden-xs">
+                  <div id="localeSelectDiv" class="pull-right hidden-xs">
```
Comment (sushain97) [inline index.html.in]: `localeSelectContainer` instead?

File: index.html.in [hunk 24]
```diff
@@ -74,7 +74,7 @@
 
         <div id="wrap">
           <nav class="navbar navbar-default" role="navigation">
-              <div class="container" style="position: relative">
+              <div class="container posRelative">
```
Comment (sushain97) [inline index.html.in]: https://getbootstrap.com/docs/4.0/utilities/position/ `position-relative` in the utilities section as discussed below.

File: assets/css/style.css [hunk 25]
```diff
@@ -103,6 +103,144 @@ html[dir='rtl'] .clearButton {
...
+#localeGlobeIcon {
+    padding: 5px 5px 0 0;
+}
+
+#srcLangSelect,
+#dstLangSelect {
+    display: inline-block !important; /* sass-lint:disable-line no-important */
+}
+
+#cancelTranslateWebpage {
```
Comment (sushain97) [inline assets/css/style.css]: Can we just add `.hide` to this instead?

File: index.html.in [hunk 26]
```diff
@@ -378,15 +378,15 @@
...
                         <li><a data-toggle="modal" data-target="#downloadModal" data-keyboard="true"><i class="fa fa-download"></i> <span data-text="Download">Download</span></a></li>
                         <li><a data-toggle="modal" data-target="#docModal" data-keyboard="true"><i class="fa fa-book"></i> <span data-text="Documentation">Documentation</span></a></li>
                         <li><a data-toggle="modal" data-target="#contactModal" data-keyboard="true"><i class="fa fa-envelope"></i> <span data-text="Contact">Contact</span></a></li>
                     </ul>
                     <div class="pull-right well well-sm helpMessage">
                         <span data-text="Notice_Mistake">Notice a mistake?</span>
-                        <a data-toggle="modal" data-target="#aboutModal" data-text="Help_Improve" data-keyboard="true" style="cursor: pointer">Help us improve Apertium!</a>
+                        <a data-toggle="modal" data-target="#aboutModal" data-text="Help_Improve" data-keyboard="true" class="pointerCursor">Help us improve Apertium!</a>
```
Comment (sushain97) [inline index.html.in]: I think this could be `. helpMessage a`?

File: index.html.in [hunk 27]
```diff
@@ -396,7 +396,7 @@
                 </div>
             </div>
 
-            <div class="pull-right well well-sm visible-xs helpMessage" style="margin-top: 10px;">
+            <div class="pull-right well well-sm visible-xs helpMessage" id="helpMessage">
```
Comment (sushain97) [inline index.html.in]: `#helpMessageXS` probably makes the most sense since there are multiple

File: index.html.in [hunk 28]
```diff
@@ -368,7 +368,7 @@
               </div>
               <div class="visible-xs">
                   <select class="localeSelect pull-left"><option> </option></select>
-                  <i class="fa fa-globe fa-2x pull-left localeGlobe" style="padding: 5px 0px 0px 5px"></i>
+                  <i class="fa fa-globe fa-2x pull-left localeGlobe" id="localeGlobeIconXs"></i>
```
Comment (sushain97) [inline index.html.in]: Do you mind switching this to `XS` instead of `Xs`? The latter looks really strange...

File: assets/css/style.css [hunk 29]
```diff
@@ -311,13 +449,16 @@ html[dir='rtl'] #srcLanguages {
     margin-right: 5px;
 }
 
+
```
Comment (sushain97) [inline assets/css/style.css]: I'm still a little confused as to why all these extra lines were added.

File: assets/css/style.css [hunk 30]
```diff
@@ -103,6 +103,144 @@ html[dir='rtl'] .clearButton {
...
+#srcLangSelect,
+#dstLangSelect {
+    display: inline-block !important; /* sass-lint:disable-line no-important */
+}
+
+#cancelTranslateWebpage {
+    display: none;
+}
+
+#translateDoc {
```
Comment (sushain97) [inline assets/css/style.css]: This rule along with all the file stuff should probably be in the translation section (there's another issue to split all this CSS into separate files, I believe)

File: index.html.in (hunk 22 above)
Comment (idealadarsh) [reply index.html.in]: I didn't understand.

Comment (sushain97) [reply index.html.in]: My question is why you are calling it that.

**Comment by sushain97** [comment]
I merged this and approved it so that I could cleanup without doing a lot of back and forth. The new code is in this branch (https://github.com/goavki/apertium-html-tools/tree/inline-styles) which you should now PR to. I have some concerns:

- you need to be a little more detail-oriented especially in a task like this one where the primary purpose is code cleanup. when a mentor asks for something to be done uniformly, please don't do it in only one instance of the problem (e.g. moving rule sets into the correct sections, camelCase'ing things). Take a look at my commit to see what kind of cleanup would be good
- you need to actually test the interface! a lot of the rules I told you to add like `position-relative` don't exist in the BS3 world (the one we're in). therefore, your changes broke lots of things (e.g. "Apertium" is randomly positioned to the left of the logo instead of to the right
